### PR TITLE
Chatform improvements

### PR DESCRIPTION
**Pull request by antis81**
# :exclamation: For reference only! Don't merge this!

**Copied description from original PR #2013 by @TheSpiritXIII** 

I've made a bunch of changes. Listen up!

Changes I’ve made:
- You can now adjust the left and right columns of the chat form by dragging them. This will surely help when text size is able to be changed. Helps with #1288 #1974 
- **moved to #2752:** ~~Hovering over text on the chat form gives the I-beam cursor.~~
- **moved to #2751:** ~~You can now search and find text in the chat form using Ctrl+F.~~
- Whitespace bug is now fixed.
- Option to show group peer list on the side instead of on top. #677
- Chats always load with some history no matter how far back it was (previously, it would load history up to 7 days back). History is loaded by amount instead of days so it never loads too much or too little.
- History is loaded dynamically when scrolling up.
- The date of the messages you’re seeing are always shown on top. You don’t have to scroll down or up to see which day you’re looking at.
- Whenever the day changes, you will see a new message about it on the chat form when you receive your next message (previously, you would have to restart, which sounded like a bug).
- Fix missing psuedo Arabic characters. #1887 
- Notification when a new message has been made while you scrolled up. #1905

### CODE REVIEW

**Comment by ovalseven8** [comment]
For the sake of completeness, here the open issues from the old PR:

---
- [ ] Font is broken (Windows) -> https://github.com/tux3/qTox/pull/2013#issuecomment-122373728
- [ ] Highlight the own user in peer list (maybe bold?)
- [ ] Another bug:
  - If you're in a groupchat the peer list is sorted alphabetically
  - Go to the Tox settings and deactivate 'Show group peer list on the side'
  - Activate it again
  -  > Peer list isn't sorted anymore
- [ ] Searching peers is not mature yet -> https://github.com/tux3/qTox/pull/2013#issuecomment-134936714
- [ ] Remove top view of peers in groupchats
- [ ] Review code and fix small code style issues

**Comment by antis81** [comment]
Thanks @ovalseven8.

@agilob I guess this will become another rewrite and a whole bunch of fun. :smile:

File: src/widget/form/groupchatform.h [hunk 1]
```diff
@@ -38,6 +41,9 @@ class GroupChatForm : public GenericChatForm
 
     void onUserListChanged();
     void peerAudioPlaying(int peer);
+    void resetLayout();
+
+    virtual void show(Ui::MainWindow &ui);
```
Comment (antis81) [inline src/widget/form/groupchatform.h]: @agilob Here's the line where it crashes, but I don't get yet why… When looking into implementation you'll find this is complete crap and the first thing we need to clean up.

**Comment by antis81** [comment]
@agilob Ok, made that compile again. Anyway the code requires a complete overhaul and is in bad shape.

**Comment by ghost** [comment]
Is this testeable already? If yes, should it be locally merged to master first or doest it work on its own?

**Comment by agilob** [comment]
Just clone that branch and compile, it works.

**Comment by ghost** [comment]
:tada: Count me in for testing!

---

> the code requires a complete overhaul and is in bad shape

From a quick run, there seem to be more bugs than a PR thread could conveniently handle. What do you think about using your issue tracker to work on branch, @antis81?

**Comment by antis81** [comment]
@kehugter Thanks! I think splitting this up into smaller chunks, which we can handle and review is the way to go.

**Comment by agilob** [comment]
We agreed with @antis81 to close it and move commit to smaller PRs, some are already awaiting.

**Pull request closed without merging**